### DISCUSSION TITLE
Is there a way to embbed a Dashboard to an external website?

### DISCUSSION THREAD

**Discussion by bruno-berchielli** [lightdash/lightdash#2739] [Q&A]
Hello,

Congrats on Lightdash, it´s an awasome project with great new ideas!

I'm evaluating it to use as a embedd solution to an existing website. The idea is:

- Create the views and dashboards in Lightdash
- Generate an embbed link with view only powers
- Use an iframe or something similar to embbed the dashboard where the end user would have a transparent experience. 

Is something like this possible or in the road map?
I searched the docs and discussions but was unable to find an answer for this. Sorry if it was discussed earlier. 

Thank you

**Comment by TuringLovesDeathMetal** [comment]
Hey @bruno-berchielli ! 

thanks so much 😊 

We haven't built embedding using Lightdash yet, but it's on our roadmap (you can see the [public roadmap here](https://github.com/orgs/lightdash/projects/7/views/1)). We're actually thinking about better reporting + sharing in Lightdash this quarter, so this issue definitely fits in there - so I'd give an estimate of having _something_ available to users in the next 3-4 months?

Is there another tool where the embedding functionality _**just worked**_ for you? (I'd love to hear about it 👀)

[This was one of the approaches a user suggested](https://github.com/lightdash/lightdash/issues/1033) - is this missing any important features for you?

**Reply by bruno-berchielli** [reply]
Hello @TuringLovesDeathMetal 

Thank you so much for this great answer!
That's great to hear, I really think this could be an awesome addition to Lightdash and would be happy to be the one of the first to try it out in the real world. 

Currently we are using Google Data Studio as a embbed solution. It's very limited in the embbed functionality. But the main reason that we are moving away of it is because it experiencing contant downtimes.

We decided to go with something open source that we can contribute with and also to host in our own servers.

Recently we tried Metabase. The embbed functionality allows for passing query params that are applied directly to the SQL queries, witch is great. Also, it allows for generation of authenticated links for embbed through the back-end, while Google Data Studio only have authentication using user/password inside the iframe or a fully public URL which is not greate at all in a security perspective.

Hope this could give Lightdash some insight! 

Thank you

**Reply by crmne** [reply]
Hi @TuringLovesDeathMetal,

I second @bruno-berchielli as we have the exact same use case. Happy to test it out too!

**Comment by grepatel** [comment]
Is there any update on this feature? I dont see it in the public roadmap. Thanks.

**Comment by mike-weinberg** [comment]
Hi @TuringLovesDeathMetal & @owlas ! 

I asked one of our UX folks at my current employer for thoughts on what good embedding would look like and one of the things they said which would be important in making an embedded product work well is ... customizable fonts (and UI elements like drop downs). 

I hadn't considered fonts to be related to embedding, but... maybe, it is. If the embedded product could inherit styling (fonts, colors, etc) from the host website either through manual configuration or (ideally) something automated, would be a huge leg up for Lightdash in an embedded use case over its competitors. I'm not sure if this is easy to do in iframes, and I'm not sure if there are non-iframe alternatives which would make this easy (a _very cursory_ glance at echarts docs suggest css inheritance is a thing), but I wanted to at least share the suggestion in case it influences your thinking on this.

_Is this worth the effort???_ I cannot claim to know, but my rationale goes like: 
> If one BI tool was loved equally by analytics engineers, business analysts, and UX designers, that tool would have a significantly stronger referral network and easier sales process than most competitors and would be awfully hard to compete with in enterprise.
